dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/encoders/EncoderRegistry.java

```diff
@@ -0,0 +1,137 @@
+/*******************************************************************************
+ * Copyright (c) 2025 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.mcp.internal.encoders;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
+import io.openliberty.mcp.content.Content;
+import io.openliberty.mcp.content.ContentEncoder;
+import io.openliberty.mcp.internal.McpCdiExtension;
+import io.openliberty.mcp.messaging.Encoder;
+import io.openliberty.mcp.tools.ToolResponse;
+import io.openliberty.mcp.tools.ToolResponseEncoder;
+import jakarta.enterprise.inject.spi.CDI;
+import jakarta.json.bind.Jsonb;
+
+public class EncoderRegistry {
+
+    private static final TraceComponent tc = Tr.register(EncoderRegistry.class);
+    private static EncoderRegistry staticInstance = null;
+
+    public static EncoderRegistry get() {
+        if (staticInstance != null) {
+            return staticInstance;
+        }
+        return CDI.current().select(McpCdiExtension.class).get().getEncoderRegistry();
+    }
```

Comment:
This pattern is slow (it requires doing a stack walk) so we should avoid using it for things that are called regularly. It should have Javadoc to this effect, or we should remove it.